### ISSUE TITLE
Last cell of a range that contains same formula in all the cells always fails to be recalculated

### ISSUE THREAD

**Issue by iulianmone16** [ClosedXML/ClosedXML#1876]
## Read and complete the full issue template

Do not randomly delete sections. They are here for a reason.

**Do you want to request a *feature* or report a *bug*?**
- [x] Bug
- [ ] Feature
- [ ] Question

**Did you test against the latest CI build?**
- [x] Yes
- [ ] No

If you answered `No`, please test with [the latest development build](https://ci.appveyor.com/project/ClosedXML/ClosedXML/branch/develop/artifacts) first.

**Version of ClosedXML**
0.95.4
e.g. 0.95.3

**What is the current behavior?**

Complete this.
When trying to read from Excel and copy the read range to another workbook, the last cell of the read range is not updated (recalculated) correctly. I have a range that contains this Excel formula : "=TIMEVALUE(TEXT(NOW(),"h:mm:ss AM/PM"))". The idea is I have 6 cells, A2:A7, and I am trying to read from it and copy the exact content to another workbook. The problem is all the cells except the last one gets recalculated in the output, the last one is copied as in the input and I have noticed that the NeedRecalculation flag on the IXLCell is always false on the last cell. The strange behaviour is although I enter the formula let's say in A8 so A7 is no longer the last cell in the range, now the A7 gets recalculated correctly in the output but A8 not. Same behaviour if I enter again in A9, A8 gets updated and A9 not. So the problem is always with the last cell in the range.

**What is the expected behavior or new feature?**

Complete this.
I expect that all the cells in the range get recalculated in the output correctly.

**Is this a regression from the previous version?**

Regressions get higher priority. Test against the latest build of the previous minor version. For example, if you experience a problem on v0.95.3, check whether it the problem occurred in v0.94.2 too. 
Yes, this is a regression.

## Reproducibility
**This is an important section. Read it carefully. Failure to do so will cause a 'RTFM' comment.**

Without a code sample, it is unlikely that your issue will get attention. Don't be lazy. Do the effort and assist the developers to reproduce your problem. Code samples should be [minimal complete and verifiable](https://stackoverflow.com/help/mcve). Sample spreadsheets should be attached whenever applicable. Remove sensitive information.

**Code to reproduce problem:**
```c#
public void Main()
{
    // Code standards:
    // - Fully runnable. I should be able to copy and paste this code into a 
    //   console application and run it without having to edit it much.
    // - Declare all your variables (this follows from the previous point)
    // - The code should be a minimal code sample to illustrate issue. The code 
    //   samples on the wiki are good examples of the terseness that I want. Don't
    //   post your full application.

// Here is a console application fully runnable. I leave the Excel workbook as an attachement.
using ClosedXML.Excel;

var workbook = new XLWorkbook("./original.xlsx");
IXLWorksheet ws;
if (workbook.TryGetWorksheet("Test", out ws))
{
    ws = workbook.Worksheet("Test");
}
var range = ws.Range("A2:A7");
object[,]? values = Array.CreateInstance(typeof(object), new int[] { range.RowCount(), range.ColumnCount() }, new int[] { 1, 1 }) as object[,];

for (int i = 1; i <= values.GetLength(0); i++)
{
    for (int j = 1; j <= values.GetLength(1); j++)
    {
        if (range.Cell(i, j) != null)
        {
            values[i, j] = range.Cell(i, j).Value;
        }
        else
        {
            values[i, j] = string.Empty;
        }
        Console.WriteLine(values[i, j]);
    }
}
}
```

- [x] I attached a sample spreadsheet.  (You can drag files on to this issue)
[original.xlsx](https://github.com/ClosedXML/ClosedXML/files/9809539/original.xlsx)

**Comment by jahav** [comment]
@iulianmone16 You are writing 
> and I am trying to read from it and copy the exact content to another workbook. 

In that case, I think you should use `IXLCell.CachedValue` property instead of `Value`. The Value will return most up to date value (e.g. your original.xlsx contains 0.40552083333333333, but Value will return the DateTime.Now of a time you requested a cell value). 

If you need to recalculate the everything on load, use `RecalculateAllFormulas`
> var workbook = new XLWorkbook(@"c:\temp\original.xlsx", new LoadOptions { RecalculateAllFormulas = true});

Sidenote: 
Since 0.97, you have to set XLDataType to XLDataType.TimeSpan or XLDataType.DateTime to have a number serial date-time from NOW() displayed as date.

Back to the root of the issue: How to deal with the volatile functions like NOW()... Let's look at the docs:
> Excel reevaluates cells that contain volatile functions, together with all dependents, every time that it recalculates. 
>                       -- https://learn.microsoft.com/en-us/office/client-developer/excel/excel-recalculation

So basically a formula with a volatile function should always be in a need of recalculation. That seems sensible and also how Excel works. Though honestly, the whole NeedsRecalculate and precedent cells need some TLC.
